packages/components/src/sites-table/sites-table-filter-tabs.tsx

```diff
@@ -0,0 +1,50 @@
+import { TabPanel } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import type { SiteData } from 'calypso/state/ui/selectors/site-data'; // eslint-disable-line no-restricted-imports
+
+interface SitesTableFilterTabsProps {
+	allSites: SiteData[];
+	children( filteredSites: SiteData[] ): JSX.Element;
+	className?: string;
+}
+
+export function SitesTableFilterTabs( {
+	allSites,
+	children: renderContents,
+	className,
+}: SitesTableFilterTabsProps ) {
+	const { __ } = useI18n();
+
+	return (
+		<TabPanel
+			className={ className }
+			tabs={ [
+				{ name: 'all', title: __( 'All', __i18n_text_domain__ ) },
+				{ name: 'launched', title: __( 'Launched', __i18n_text_domain__ ) },
+				{ name: 'private', title: __( 'Private', __i18n_text_domain__ ) },
+				{ name: 'coming-soon', title: __( 'Coming soon', __i18n_text_domain__ ) },
+			] }
+		>
+			{ ( tab ) => renderContents( filterSites( allSites, tab.name ) ) }
+		</TabPanel>
+	);
+}
+
+function filterSites( sites: SiteData[], filterType: string ): SiteData[] {
+	return sites.filter( ( site ) => {
+		const isComingSoon =
+			site.is_coming_soon || ( site.is_private && site.launch_status === 'unlaunched' );
```

Comment:
Something here is not loading my "coming soon" sites.
I'll fix it in my PR, when I add the tabs.

Reply:
Yeah, this logic is a bit of a mess 😅 I tried to copy the logic that shows the `[Coming Soon]` and `[Private]` badges in the site switcher but I think I messed it up. I figured it was best to just fix it up in a later PR, so thanks 🙂